android-app/app/src/main/java/de/dhbw/ka/se/fibo/CreateAccountActivity.java

```diff
@@ -55,9 +103,89 @@ private void initializeButton() {
     }
 
     private void registerButtonClicked() {
-        Intent i = new Intent(CreateAccountActivity.this,
+        i = new Intent(CreateAccountActivity.this,
                 MainActivity.class);
-        startActivity(i);
-        finish();
+        create_account_email = findViewById(R.id.create_account_email);
+        String email = create_account_email.getText().toString();
+        passwordField = findViewById(R.id.create_account_password_layer);
+        emailField = findViewById(R.id.create_account_email_layer);
+        create_account_password = findViewById(R.id.create_account_password);
+        String password = create_account_password.getText().toString();
+        createUser(email, password);
+
+    }
+
+    private void createUser(String email, String password) {
+
+        if (!checkValidInput(email, password)) {
+            return;
+        }
+
+        RequestQueue queue = Volley.newRequestQueue(this);
+        StringRequest request = new StringRequest(Request.Method.POST, url, response -> {
+
+            Toast successToast = Toast.makeText(this, "Success", Toast.LENGTH_LONG);
+            successToast.setGravity(Gravity.TOP, 0, 0);
+            successToast.show();
+            System.out.println("Successfully" + response);
+            startActivity(i);
+            finish();
```

Comment:
```suggestion
            Intent i = new Intent(this,MainActivity.class);
            startActivity(i);
            finish();
```

Here you can add the intent again. In that way it will not be created without any need